src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -242,48 +242,52 @@ private static void HandleLastWin32Error()
 
     internal static class NtProcessInfoHelper
     {
-        // Cache a single buffer for use in GetProcessInfos().
-        private static long[]? CachedBuffer;
-
         // Use a smaller buffer size on debug to ensure we hit the retry path.
 #if DEBUG
-        private const int DefaultCachedBufferSize = 1024;
+        private const int DefaultCachedBufferSize = 8 * 1024;
 #else
-        private const int DefaultCachedBufferSize = 128 * 1024;
+        private const int DefaultCachedBufferSize = 1024 * 1024;
```

Comment:
The starting buffer size should be much lower. It is ok to retry for the first time on a machine with very large number of processes.

Reply:
I did not change the value. It was `128 * 1024` before but it was later allocating an array of `long`s so this is why it's `x8` more now`.

What do you suggest? Half of it? 1/4?

Reply:
I have re-read your other post and I see that we need ~0.6MB buffer on a typical machine. I though it is much more.

Please gnore this feedback.